### DISCUSSION TITLE
Semi-regular grid interpolation to point observations

### DISCUSSION THREAD

**Discussion by davidkuhl** [dtcenter/METplus#1537] [Configuration] [answered]
In the MET point_stat tutorials there are examples where you read in regular grids in 4-dimensions: Lat x Lon x Lev x Time as well as irregular observations as points.  Then you interpolate the regular grid onto the point observations.

My question is: what if I have a regular grid in 3-dimensions: Lat x Lon x Time but it is irregular in Lev, i.e. at any Lat x Lon point the levels are different.  I still want to read in irregular observations and interpolate the semi-regular grid to the observations.  Is there an example of how to read in semi-regular gridded data?

I apologize if this is obvious, I am new to the METplus suite of codes.

Thanks,
Dave.

**Comment by j-opatz** [comment, marked as answer]
Hi Dave,

And thanks for your question. I'm going to tag @JohnHalleyGotway for a more definitive answer, but he's out of the office currently.

What it sounds like you're doing is vertical interpolation of levels, would that be a correct summary? We had a (somewhat) similar discussion [here](https://github.com/dtcenter/METplus/discussions/1185) on that.  At the moment, METplus does not offer vertical interpolation of levels, and requires standard pressure/height levels for grid definition.

**Reply by davidkuhl** [reply]
Yes, exactly--vertical interpolation of levels.  Thanks for the link, that is exactly what I was worried about.

thanks,
dave.
